src/app/shared/table/table.component.ts

```diff
@@ -25,6 +26,7 @@ import { TableDataSource } from './table-data-source';
 export class TableComponent implements OnInit, AfterViewInit, OnDestroy {
   @Input() public dataSource!: TableDataSource<TableData>;
   @ViewChild('searchInput') public searchInput!: ElementRef;
+  @ViewChild(DaterangepickerDirective) public picker: DaterangepickerDirective;
```

Comment:
If you have 2 date time range picker control in the template based on different props (start/endDate) how do you handle them with one picker?

Reply:
We can't handle it right now as both filters will have the same id. We update the filter values based on id. We need to make the id as an argument that can be modified during filter creation. Additional changes might also be required. As this piece of code is necessary for the calendar icon which on click must open the picker, I have removed the calendar icon. By default as the value is displayed, addition of a calendar icon might not be necessary.

Reply:
You can retrieve a list of elements in Angular:
https://angular.io/api/core/ViewChildren
Could you check?

Reply:
Tried with multiple inputs. It works. Removed the second element after trial.
![image](https://user-images.githubusercontent.com/5779016/148984854-1489436b-96cc-4324-b3a8-b38b617055cc.png)